sys/cddl/dev/dtrace/mips/dtrace_isa.c

```diff
@@ -219,22 +211,14 @@ uint64_t
 dtrace_getarg(int arg, int aframes)
 {
 	int i;
-	register_t sp, ra, pc;
+	register_t pc, sp;
 	/* XXX: Fix this ugly code */
 	register_t args[8];
 	int valid[8];
 
+	pc = (intptr_t)&&here;
```

Comment:
Not 100% sure this will always do the right thing for purecap. It should do for MIPS and RISC-V but not sure about Morello.
I'm not sure whether address-of-label needs to give a valid pointer or if an offset relative to entry point is also fine. I think as long as `goto *foo;` works anything is fine.

Reply:
I do think this will need some refinement for purecap, if nothing else we will have to unseal it for at least MIPS since we read the instruction to figure out the stack layout.  The current MIPS purecap kernel has its own stack_machdep.c and doesn't support DTrace or DDB, so it hasn't really had to fully wrestle with this problem yet.  It does though already have to deal with unsealing return addresses.

Do note that the use of address-of-label is MIPS only.  RISC-V and aarch64 use different approaches.